hotham-simulator/src/state.rs

```diff
@@ -190,43 +202,94 @@ impl State {
         println!("[HOTHAM_SIMULATOR] All things are now destroyed");
     }
 
-    pub fn update_actions(&mut self) -> Option<()> {
-        let mut z_delta = 0.00;
-        let mut x_delta = 0.00;
-        let mut x_rot = 0.0;
-        let mut y_rot = 0.0;
+    /// Updates the OpenXR camera Position & Rotation
+    /// Tries to emulate a simple first person floating camera to help navigate the scene
+    pub fn update_camera(&mut self) -> Option<()> {
+        let mut x_rot = 0f32;
+        let mut y_rot = 0f32;
+
+        // We need to adjust the speed value so its always the same speed even if the frame rate isn't consistent
+        // The delta time is the the current time - last frame time
+        let now = Instant::now();
+        let delta = now - self.last_frame_time;
+        self.last_frame_time = now;
+
+        let dt = delta.as_secs_f32();
+        let movement_speed = 2f32 * dt;
+        let mouse_sensitivity = 4f32 * dt;
+
+        if let Ok(input_event) = self.event_rx.as_ref()?.try_recv() {
+            match input_event {
+                DeviceEvent::Key(keyboard_input) => self.input_state.process_event(keyboard_input),
+                DeviceEvent::MouseMotion { delta: (x, y) } => {
+                    x_rot = -x as _;
+                    y_rot = -y as _;
+                }
+                _ => {}
+            }
+        }
+
+        // Camera position & Rotation
+        let pose = &mut self.view_poses[0];
+
+        // Update Rotation
+        let orientation = &mut pose.orientation;
+
+        self.camera.yaw += x_rot * mouse_sensitivity;
+        self.camera.pitch += y_rot * mouse_sensitivity;
+
+        // I think I'm converting these two types incorrectly but it seems to work and I'm too scared to break it lol
+        let rotation: Unit<Quaternion<f32>> =
```

Comment:
If you want to go between `nalgebra` and `openxr`, you can use `mint` as an intermediary to do it safely. Take a look at this function as an example:

https://github.com/leetvr/hotham/blob/519c18e47a1784a46cebbedfd93bac6697b56db3/hotham/src/util.rs#L78

Reply:
ah makes sense, wasn't sure what mint was used for